vue2-vuetify/src/complex/components/AdditionalProperties.vue

```diff
@@ -0,0 +1,463 @@
+          messages.push(
+            `Property '${this.newPropertyName}' is already defined`
+          );
+        }
+
+        // JSONForms has special means for "[]." chars - those are part of the path composition so for not we can't support those without special handling
+        if (this.newPropertyName.includes('[')) {
+          messages.push('Property name contains invalid char: [');
+        }
+        if (this.newPropertyName.includes(']')) {
+          messages.push('Property name contains invalid char: ]');
+        }
+        if (this.newPropertyName.includes('.')) {
+          messages.push('Property name contains invalid char: .');
+        }
+
+        return messages;
+      }
+
+      return [];
+    },
+    placeholder(): string {
+      return this.t(this.i18nKey('newProperty.placeholder'), 'New Property');
+    },
+    reservedPropertyNames(): string[] {
+      return Object.keys(this.control.schema.properties || {});
+    },
+    additionalPropertiesTitle(): string | undefined {
+      const additionalProperties = this.control.schema.additionalProperties;
+
+      const label =
+        typeof additionalProperties === 'object' &&
+        Object.prototype.hasOwnProperty.call(additionalProperties, 'title')
+          ? additionalProperties.title ?? 'Additional Properties'
+          : 'Additional Properties';
+
+      return this.t(this.i18nKey('title'), label);
+    },
+    addToLabel(): string {
+      return this.t(
+        this.i18nKey('btn.add'),
+        'Add to ${additionalProperties.title}',
+        {
+          additionalProperties: {
+            title: this.additionalPropertiesTitle,
+          },
+        }
+      );
+    },
+    deleteLabel(): string {
+      return this.t(
+        this.i18nKey('btn.delete'),
+        'Delete from ${additionalProperties.title}',
+        {
+          additionalProperties: {
+            title: this.additionalPropertiesTitle,
+          },
+        }
+      );
+    },
+  },
+  watch: {
+    'control.data': {
+      handler(newData) {
+        // revert back any undefined values back to the default value when the key is part of the addtional properties since we want to preserved the key
+        // for example when we have a string additonal property then when we clear the text component the componet by default sets the value to undefined to remove the property from the object - for additional properties we do not want that behaviour
+        if (typeof this.control.data === 'object') {
+          const keys = Object.keys(newData);
+          let hasChanges = false;
+          this.additionalPropertyItems.forEach((ap) => {
+            if (
+              (!keys.includes(ap.propertyName) ||
+                newData[ap.propertyName] === undefined ||
+                newData[ap.propertyName] === null) &&
+              ap.schema
+            ) {
+              hasChanges = true;
+              newData[ap.propertyName] = createDefaultValue(ap.schema);
```

Comment:
What happens if the default value is `undefined` or `null`? Could this cause an endless loop?

Reply:
yes you are right - in the special case where the schema type is 'null' then this can happen - will add code fix for that

Reply:
resolved in the latest commit

Reply:
sure thing